hpx/runtime/threads/detail/create_work.hpp

```diff
@@ -81,12 +81,16 @@ namespace hpx { namespace threads { namespace detail
         // Pass critical priority from parent to child.
         if (self)
         {
-            if (thread_priority_critical == threads::get_self_id()->get_priority())
-                data.priority = thread_priority_critical;
+            if (thread_priority_high_recursive ==
+                threads::get_self_id()->get_priority())
+            {
+                data.priority = thread_priority_high_recursive;
```

Comment:
Right, this should additionally check whether `data.priority == thread_priority_default`. Which also means that we should make `thread_priority_default` and `thread_priority_normal` distinct in terms of enum values represented (just treat them in similar ways).